WebApi/WebApi/Models/DTO/PostDTOs/PostEditDTO.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApi.Models.DTO.PostDTOs
+{
+    //DTO for creating and editing post
+    public class PostEditDTO : IValidatableObject
+    {
+        [Required]
+        [MaxLength(50)]
+        public string title { get; set; }
+        [Required]
+        public string content { get; set; }
+        [Required]
+        public int category { get; set; }
+        [Required]
+        public bool isPromoted { get; set; }
+        [Required]
+        public DateTime dateTime { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
```

Comment:
jeżeli nie ma potrzeby jakiś customowych filtrów to można nie implementować tego interfejsu, bo nwm czy jak zwracasz null to będą przechodziły te walidacje, bo ta metoda się automatycznie wywołuje